components/data/data-sources/src/main/java/org/eclipse/dirigible/components/data/sources/manager/DataSourceInitializer.java

```diff
@@ -0,0 +1,157 @@
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.stereotype.Component;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+@Component
+public class DataSourceInitializer {
+
+    /** The Constant logger. */
+    private static final Logger logger = LoggerFactory.getLogger(DataSourceInitializer.class);
+
+    /** The Constant DATASOURCES. */
+    private static final Map<String, javax.sql.DataSource> DATASOURCES = Collections.synchronizedMap(new HashMap<>());
+
+
+    private final ApplicationContext applicationContext;
+
+    DataSourceInitializer(ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    /**
+     * Initialize data source.
+     *
+     * @param name the name
+     * @return the {@link javax.sql.DataSource}
+     */
+    public javax.sql.DataSource initialize(DataSource dataSource) {
+        String name = dataSource.getName();
+
+        if (isAlreadyInitialized(name)) {
+            return getAlreadyInitializedDataSource(name);
+        }
+
+        logger.info("Initializing a datasource with name: [{}]", name);
+        if ("org.h2.Driver".equals(dataSource.getDriver())) {
+            try {
+                prepareRootFolder(name);
+            } catch (IOException ex) {
+                logger.error("Invalid configuration for the datasource: [{}]", name, ex);
+            }
+        }
+        Properties properties = new Properties();
+        properties.put("driverClassName", dataSource.getDriver());
+        properties.put("jdbcUrl", dataSource.getUrl());
+        properties.put("dataSource.url", dataSource.getUrl());
+        properties.put("dataSource.user", dataSource.getUsername());
+        properties.put("dataSource.password", dataSource.getPassword());
+        properties.put("dataSource.logWriter", new PrintWriter(System.out));
+
+        Map<String, String> hikariProperties = getHikariProperties(name);
+        hikariProperties.forEach(properties::setProperty);
+
+        HikariConfig config = new HikariConfig(properties);
+        config.setPoolName(name);
+        config.setAutoCommit(true);
+        dataSource.getProperties()
+                  .forEach(dsp -> config.addDataSourceProperty(dsp.getName(), dsp.getValue()));
+        HikariDataSource hds = new HikariDataSource(config);
+
+        ManagedDataSource managedDataSource = new ManagedDataSource(hds);
+        registerDataSourceBean(name, managedDataSource);
+
+        DATASOURCES.put(name, managedDataSource);
+
+        return managedDataSource;
+    }
+
+    public boolean isAlreadyInitialized(String dataSourceName) {
+        return DATASOURCES.containsKey(dataSourceName);
+    }
+
+    public javax.sql.DataSource getAlreadyInitializedDataSource(String dataSourceName) {
+        return DATASOURCES.get(dataSourceName);
+    }
+
```

Comment:
let introduce method for removing already initalized datasource

Reply:
done